events/s3_batch_job.go

```diff
@@ -33,7 +45,7 @@ type S3BatchJobResponse struct {
 
 // S3BatchJobResult represents the result of a given task
 type S3BatchJobResult struct {
-	TaskID       string `json:"taskId"`
-	ResultCode   string `json:"resultCode"`
-	ResultString string `json:"resultString"`
+	TaskID       string               `json:"taskId"`
+	ResultCode   S3BatchJobResultCode `json:"resultCode"`
```

Comment:
I think adding the type creates a breaking change. 

ex: https://play.golang.org/p/T5G1eKtbBor

I think leaving the type as `string` is fine